drivers/hmis/app/models/hmis/form/definition_validator.rb

```diff
@@ -271,13 +271,15 @@ def check_cdeds(document, role)
         cded_type = cded.field_type
 
         case item_type
-        when 'GROUP', 'OBJECT', 'FILE', 'IMAGE'
+        when 'GROUP', 'OBJECT'
           # We don't expect these types to have custom field mappings. If they do, raise an error
           raise "Item #{link_id} has type #{item_type}, so it should not have a custom_field_key"
         when 'DISPLAY'
           # DISPLAY types should really be in the above category too,
           # but we have existing cases that store an autofill value
           next
+        when 'FILE', 'IMAGE'
+          raise_bad_type_match(link_id, item_type, cded_key, cded_type) unless cded_type == 'file'
```

Comment:
same question here with `.downcase`

Reply:
I think we are confident it will always be lowercase, because `cded.field_type` is generated based on which of the `value_` columns is non-nil. https://github.com/greenriver/hmis-warehouse/blob/681544cdafe35c9ec96ccc0e706a14cb371382a3/drivers/hmis/app/models/hmis/hud/custom_data_element.rb#L62

Reply:
I don't think the referenced code is applicable. On line 271 we assign `cded_type = cded.field_type` which is what's being referenced here. It looks like we maybe do still need a rails validation constraint on the `CustomDataElementDefinitions.field_type` column, the values are expected to match these:
https://github.com/greenriver/hmis-warehouse/blob/0c5bd1c35388d1f618ceea8ce23ed9e961c7886e/drivers/hmis/app/models/hmis/hud/custom_data_element_definition.rb#L14-L22

(Where you already added `file`).

I do worry that we are breaking the following assumption:
* The `CustomDataElement` has an attribute `value_[x]` where "x" matches the `CustomDataElementDefinition.field_type`.

That assumption is broken because the field name is `value_file_id` whereas the field_type is `file`. I think that could cause problems, but maybe it's masked because `value_file` is still an assignable attribute?

Reply:
Thanks Gig. I can add a validation. I think you are right about the assumption breaking, but I can't think of a good way around it. I played around with it a bit, but ran up against the fact that in the form processing code, when the value is saved, we can't assign to `value_file_id` because we don't have a file ID yet, since the file record hasn't yet been saved.

Reply:
Hm ok I'm proposing to remove this validation for now since it causes test failures. (https://github.com/greenriver/hmis-warehouse/actions/runs/12188033326/job/34000177154) This PR is already way over estimate so I'd like to descope fixing the tests.